app/services/participant_profile_resolver.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class ParticipantProfileResolver
+  class << self
+    def call(participant_identity, course_identifier)
+      return unless participant_identity
+
+      if ParticipantProfile::ECT::COURSE_IDENTIFIERS.include?(course_identifier)
+        participant_identity
+          .participant_profiles
+          .active_record
+          .ects
+          .first
```

Comment:
this code was exisiting in other modules. This is however wrong! this randomly chooses a profile we don't know which one it is. Participants are supposed to have only one ECF profile of each type at most but we know this is not the case.

Reply:
Is this problem logged in a ticket for a follow up?